### DISCUSSION TITLE
items.onBeforeCreate hook

### DISCUSSION THREAD

**Discussion by rrrnld** [directus/directus#10845] [Archive]
Is there a hook like this? The use case would be to pre-fill some nested fields, with data we'd like to copy over from another collection.

**Comment by rijkvanzanten** [comment]
See the `items.create` filter hook: https://docs.directus.io/extensions/hooks/#available-events 🙂

**Reply by rrrnld** [reply]
This is called after a new item has been saved if I understand it correctly?

**Reply by azrikahar** [reply]
The type of hook would be the keyword here. `items.create` **filter** hook is before, `item.create` **action** hook is after 👍

**Reply by rrrnld** [reply]
Thanks! :)
